src/query/ee/src/storages/fuse/operations/vacuum_temporary_files.rs

```diff
@@ -18,45 +18,159 @@ use std::time::UNIX_EPOCH;
 
 use databend_common_exception::Result;
 use databend_common_storage::DataOperator;
+use futures_util::stream;
 use futures_util::TryStreamExt;
+use opendal::Entry;
+use opendal::EntryMode;
 use opendal::Metakey;
 
 #[async_backtrace::framed]
 pub async fn do_vacuum_temporary_files(
     temporary_dir: String,
     retain: Option<Duration>,
     limit: Option<usize>,
-) -> Result<Vec<String>> {
+) -> Result<usize> {
+    let limit = limit.unwrap_or(usize::MAX);
+    let expire_time = retain
+        .map(|x| x.as_millis())
+        .unwrap_or(1000 * 60 * 60 * 24 * 3) as i64;
+    let timestamp = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_millis() as i64;
+
     let operator = DataOperator::instance().operator();
 
+    let meta_key = Metakey::Mode | Metakey::LastModified;
     let mut ds = operator
         .lister_with(&temporary_dir)
-        .recursive(true)
-        .metakey(Metakey::LastModified)
+        .metakey(meta_key)
         .await?;
 
-    let limit = limit.unwrap_or(usize::MAX);
-    let expire_time = retain.map(|x| x.as_millis()).unwrap_or(60 * 60 * 24 * 3) as i64;
-    let timestamp = SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .unwrap()
-        .as_millis() as i64;
+    let mut removed_temp_files = 0;
+
+    while removed_temp_files < limit {
```

Comment:
It would be better to add some `info!` logs in this function. 
For instance, logging the duration of the `vacuum temp files` process, which took an hour when I executed it, but don't know what progress is going.